admin/marketplace_admin/commands/renew.py

```diff
@@ -0,0 +1,83 @@
+# Copyright 2017 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+import logging
+from marketplace_admin.services import api
+from marketplace_admin.services import data
+
+
+LOGGER = logging.getLogger(__name__)
+
+
+def init_renew_parser(subparsers):
+    parser = subparsers.add_parser(
+        'renew',
+        help='Closes and reopens any "renewable" Offers',
+        parents=[api.get_parser(), data.get_parser()])
+    return parser
+
+
+def do_renew(opts):
+    LOGGER.info('Reading data: %s', opts.data)
+    renew_data = data.load(opts.data)
+
+    LOGGER.info('Submitting data to URL: %s', opts.url)
+    fetch = lambda path: api.get(opts.url, path)
+    submit = lambda p, b, a=None: api.post(opts.url, p, b, a)
+    update = lambda p, b, a: api.patch(opts.url, p, b, a)
```

Comment:
Not exactly the pythonic way, but interesting. Looks `functools.partial` would solve this

Reply:
Being pythonic is no longer something I am particularly concerned with. Switched to partial.